File: hello_world_and_crash.cpp
/*
* Hello World and Crash for Turtlebot
* Moves Turtlebot forward until you ctrl + c
* Tested using TurtleBot 2, ROS Indigo, Ubuntu 14.04
*/

#include <iostream>
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>

int main(int argc, char** argv)
{
    //init the ROS node
    ROS_INFO_STREAM("Hello World");
    ros::init(argc, argv, "robot_driver");
    ros::NodeHandle nh;

    //init publisher
    ros::Publisher cmd_vel_pub_;
    cmd_vel_pub_ = nh.advertise<geometry_msgs::Twist>("cmd_vel_mux/input/teleop", 1);
    
    //init direction that turtlebot should go
    geometry_msgs::Twist base_cmd;
    base_cmd.linear.x = base_cmd.linear.y = base_cmd.angular.z = 0;

    //and let's go forward by setting X to a positive value
    base_cmd.linear.x = 0.25;

    ROS_INFO_STREAM("And Crashing ... ctrl + c to stop me :)");

    while(nh.ok()) { //have we ctrl + C?  If no... keep going!
        //"publish" sends the command to turtlebot to keep going
    	cmd_vel_pub_.publish(base_cmd);
    }
}

File: hello_world_and_crash.py
#!/usr/bin/env python

import rospy
from geometry_msgs.msg import Twist

class LetsCrash():
    def __init__(self):
        # initiliaze
        rospy.init_node('lets_crash', anonymous=False)

        # What to do you ctrl + c    
        rospy.on_shutdown(self.shutdown)
        
        # You may need to change cmd_vel_mux/input/teleop to /cmd_vel if you're not using TurtleBot2
        self.cmd_vel = rospy.Publisher('cmd_vel_mux/input/teleop', Twist)
     
        r = rospy.Rate(10);

        # let's go forward at 0.2 m/s
        move_cmd = Twist()
        move_cmd.linear.x = 0.2

        while not rospy.is_shutdown():
            self.cmd_vel.publish(move_cmd)
            r.sleep()
                        
        
    def shutdown(self):
        # stop turtlebot
        rospy.loginfo("Stop Crashing")
        self.cmd_vel.publish(Twist())
        rospy.sleep(1)
 
if __name__ == '__main__':
    try:
        LetsCrash()
    except:
        rospy.loginfo("Let's crash node terminated.")

